docs/extension-development/working-with-woocommerce-admin-pages.md

```diff
@@ -0,0 +1,119 @@
+* `nav_args` (_optional_) — An array of parameters for registering items in WooCommerce Navigation. (see usage below)
+    * `order` – Order number for presentation.
+    * `parent` – Menu for item to fall under. For example: `woocommerce`, `woocommerce-settings` or `woocommerce-analytics`. Categories added by an extension are available as well.
+    * `parentPath` - The path to the parent page (the `path` value of the query arg for the parent page's url) to highlight in the menu when this page is active. This is useful for pages that are not direct children of a WooCommerce menu item.
+
+Registering a React-powered page is similar to connecting a PHP page, but with some key differences. Registering pages will automatically create WordPress menu items for them, with the appropriate hierarchy based on the value of `parent`.
+
+### Example: Adding a new WooCommerce Admin page
+
+```php
+function add_extension_register_page() {
+    if ( ! function_exists( 'wc_admin_register_page' ) ) {
+        return;
+    }
+ 
+    wc_admin_register_page( array(
+        'id'       => 'my-example-page',
+        'title'    => 'My Example Page',
+        'parent'   => 'woocommerce',
+        'path'     => '/example',
+        'nav_args' => array(
+            'order'  => 10,
+            'parent' => 'woocommerce',
+        ),
+    ) );
+}
+add_action( 'admin_menu', 'add_extension_register_page' );
+```
+
+In the example above, we encapsulated our call to [`wc_admin_register_page()`](https://woocommerce.github.io/code-reference/namespaces/default.html#function_wc_admin_register_page) in a function that we have hooked to the [`admin_menu`](https://developer.wordpress.org/reference/hooks/admin_menu/) action. Once you have registered a page with the controller, you can supply a React component on the client side.
+
+```js
+import { addFilter } from '@wordpress/hooks';
+ 
+const MyExamplePage = () => <h1>My Example Extension</h1>;
+ 
+addFilter( 'woocommerce_admin_pages_list', 'my-namespace', ( pages ) => {
+    pages.push( {
+        container: MyExamplePage,
+        path: '/example',
+        breadcrumbs: [ 'My Example Page' ],
+        navArgs: {
+            id: 'my-example-page',
+        },
+    } );
+ 
+    return pages;
+} );
+```
+
+Above, we’re creating a simple [functional React component](https://reactjs.org/docs/components-and-props.html#function-and-class-components) for the sake of demonstration, but a real-world extension would likely have a more complex nesting of components.
+
+When supplying a component to the list of WooCommerce Admin Pages, it’s important to make sure that the value you specify for `navArgs.id` matches the `id` for the page you register with `PageController` in your call to [`wc_admin_register_page()`](https://woocommerce.github.io/code-reference/namespaces/default.html#function_wc_admin_register_page).
+
```

Comment:
Since there is no section that explains the Javascript part, either:
1. We need to add that section explaining the object structure to be appended to `pages`, that is:
```
{
    container: MyExamplePage,
    path: '/example',
    breadcrumbs: [ 'My Example Page' ],
    navArgs: {
        id: 'my-example-page',
        parentPath: '/other-example'
    },
}
```
or

2. Have an extra paragraph to explain `navArgs.parentPath` just like we have a paragraph above explaining `navArgs.id`. Here's what I can come up with, feel free to edit:

> Pass the path to the parent page (the `path` value of the query arg for the parent page's url) as `navArgs.parentPath` to highlight that parent page menu when this page is active.  For example:

```
{
    container: TransactionDetailsPage,
    path: '/payments/transactions/details',
    breadcrumbs: [ 'Payments', 'Transactions', 'Transaction Details' ],
    navArgs: {
        id: 'transaction-details',
        parentPath: '/payments/transactions'
    },
}
```

Reply:
Done. I think that ultimately this doc could be expanded upon further to better explain and document the details of what is going on, but I have added `parentPath` and think it is better to get a base version of this doc migrated than wait for more in-depth expansion.